CMakeLists.txt

```diff
@@ -1,94 +1,242 @@
-option(KOMPUTE_OPT_BUILD_DOCS "Enable if you want to build documentation" 0)
-option(KOMPUTE_OPT_BUILD_SHADERS "Enable if you want to re-build all shader files" 0)
-option(KOMPUTE_OPT_BUILD_SINGLE_HEADER "Enable if you want to build the single header file" 0)
-option(KOMPUTE_OPT_INSTALL "Enable if you want to enable installation" 0)
+kompute_option(KOMPUTE_OPT_BUILD_TESTS "Enable if you want to build tests." OFF)
+kompute_option(KOMPUTE_OPT_CODE_COVERAGE "Enable if you want code coverage." OFF)
+kompute_option(KOMPUTE_OPT_BUILD_DOCS "Enable if you want to build documentation." OFF)
+kompute_option(KOMPUTE_OPT_INSTALL "Enable if you want to enable installation." OFF)
+
 # Build options
-option(KOMPUTE_OPT_BUILD_PYTHON "Enable if you want to build python bindings" 0)
-option(KOMPUTE_OPT_ENABLE_SPDLOG "Enable to compile with spdlog as the internal logging framework" 0)
-option(KOMPUTE_OPT_REPO_SUBMODULE_BUILD "Use the submodule repos instead of external package manager" 0)
-option(KOMPUTE_OPT_ANDROID_BUILD "Enable android compilation flags required" 0)
-option(KOMPUTE_OPT_DISABLE_VK_DEBUG_LAYERS "Explicitly disable debug layers even on debug" 0)
-option(KOMPUTE_OPT_DEPENDENCIES_SHARED_LIBS "Whether to use shared libraries for dependencies for install" 0)
-option(KOMPUTE_OPT_BUILD_AS_SHARED_LIB "Whether to build kompute as shared library" 0)
-# Build flags
-set(KOMPUTE_EXTRA_CXX_FLAGS "" CACHE STRING "Extra compile flags for Kompute, see docs for full list")
-
-if(KOMPUTE_OPT_ENABLE_SPDLOG)
-    set(KOMPUTE_EXTRA_CXX_FLAGS "${KOMPUTE_EXTRA_CXX_FLAGS} -DKOMPUTE_ENABLE_SPDLOG=1")
-    set(SPDLOG_FMT_EXTERNAL ON CACHE BOOL "Enables external fmt as its current dep" FORCE)
-    if(KOMPUTE_OPT_INSTALL)
-        # Enable install parameters for spdlog (overrides parameters passed)
-        set(SPDLOG_INSTALL ON CACHE BOOL "Enables install of spdlog" FORCE)
-
-        if(KOMPUTE_OPT_DEPENDENCIES_SHARED_LIBS)
-            set(SPDLOG_BUILD_SHARED ON CACHE BOOL "Enables build of shared libraries" FORCE)
-        endif()
+kompute_option(KOMPUTE_OPT_BUILD_PYTHON "Enable if you want to build python bindings." OFF)
+kompute_log_level(KOMPUTE_OPT_LOG_LEVEL "Internally we use spdlog for logging. The log level used can be changed here. Possible values: 'Trace', 'Debug', 'Info', 'Warn', 'Error', 'Critical', 'Off', 'Default'. If set to 'Off' spdlog will be deactivated completely. If set to 'Default', the log level will be set to 'Info' for release builds and 'Debug' else." "Default")
+kompute_option(KOMPUTE_OPT_ANDROID_BUILD "Enable android compilation flags required." OFF)
+kompute_option(KOMPUTE_OPT_DISABLE_VK_DEBUG_LAYERS "Explicitly disable debug layers even on debug." OFF)
+kompute_option(KOMPUTE_OPT_DISABLE_VULKAN_VERSION_CHECK "Whether to check if your driver supports the Vulkan Header version you are linking against. This might be useful in case you build shared on a different system than you run later." OFF)
```

Comment:
Separate to the point above, I found that even if I configure the display for vulkaninfo to work, there is still an issue with the build, which provides the following error:

```
-- =======================================================
-- Ensuring the currently installed driver supports the Vulkan version requested by the Vulkan Header.
-- Found Vulkan Header version 1.2.203.
-- GPU0 supports Vulkan API version '1.1.0'.
-- GPU0 does not support Vulkan > 1.1.0.
CMake Error at cmake/check_vulkan_version.cmake:126 (message):
  None of your GPUs supports Vulkan Header 1.2.203.  Please try updating your
  driver, or downgrade your Vulkan headers.
Call Stack (most recent call first):
  CMakeLists.txt:132 (check_vulkan_version)


-- Configuring incomplete, errors occurred!
See also "/home/alejandro/Programming/kubernetes/seldon/vulkan-kompute/build/CMakeFiles/CMakeOutput.log".
Makefile:57: recipe for target 'mk_cmake' failed
make: *** [mk_cmake] Error 1
```

I am using swiftshader so want to validate if indeed this is because of this issue?

Reply:
I extended the comment printed when this check fails. But in my eyes this is intended to fail here since I by now run into multiple issues where my driver was to old compared to my Vulkan header especially related to Vulkan Debug Layers.

Reply:
Yeah it's quite strange, it seems that it doesn't really like SWIFTSHADER, as that's what it seems to fail on - based on this my thinking is that it could be better to have it disabled. Mainly thinking to reduce friction as it does seem like this check fails in cases where it should pass - however your point is valid about "it shold fail and running it should be explicit", do you have more details on then what would be the fix? Is this suggesting that I should update the current driver to 1.1.203+ (and somehow swiftshader should be updated as well)?